server/datastore/mysql/maintained_apps.go

```diff
@@ -53,9 +53,12 @@ ON DUPLICATE KEY UPDATE
 		// upsert the maintained app
 		res, err := tx.ExecContext(ctx, upsertStmt, app.Name, app.Token, app.Version, app.Platform, app.InstallerURL,
 			app.SHA256, app.BundleIdentifier, installScriptID, uninstallScriptID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "upsert maintained app")
+		}
 		id, _ := res.LastInsertId()
 		appID = uint(id) //nolint:gosec // dismiss G115
-		return ctxerr.Wrap(ctx, err, "upsert maintained app")
```

Comment:
This would wrap and return the error if present, no? But I agree it looks better with the explicit check

Reply:
I think the problem would be that if there's an error, `res` is nil and so on line 56 there'd be a panic

Reply:
Ah sorry, didn't see Jahziel's comment before I replied.

Reply:
Gotcha!